app/controllers/election.js

```diff
@@ -17,12 +18,41 @@ exports.load = (req, res, next, electionId) =>
     });
 
 exports.retrieveActive = (req, res) =>
-  Election.findOne({ active: true, hasVotedUsers: { $ne: req.user._id } })
+  Election.findOne({
+    active: true,
+    hasVotedUsers: { $ne: req.user._id },
+  })
     .select('-hasVotedUsers')
     .populate('alternatives')
     .exec()
     .then((election) => {
-      res.status(200).json(election);
+      // There is no active election (that the user has not voted on)
+      if (!election) {
+        return res.sendStatus(404);
+      }
+      // If the user is active, then we can return the election right
+      // away, since they have allready passed the access code prompt
+      if (req.user.active) {
+        return res.status(200).json(election);
+      }
+      // There is an active election that the user has not voted on
+      // but they did not pass any (or the correct) access code,
+      // so we return 403 which prompts a access code input field.
+      else if (
```

Comment:
Can you change this to an `if` instead, since you are returning? I think that flow makes more sense and makes it easier to read. You can then also add some empty lines to get some more spacing.